File: diffapp/test/ci_analyze_build_jobs_test.dart
import 'dart:io';

import 'package:flutter_test/flutter_test.dart';

void main() {
  test('CI に解析/テスト/ビルド用のジョブが定義されている', () async {
    // リポジトリルートのワークフロー定義を読む（テストのCWDは diffapp/）
    const path = '../.github/workflows/flutter-ci.yml';
    final file = File(path);
    expect(file.existsSync(), isTrue, reason: 'flutter-ci.yml が存在すること');

    final yml = await file.readAsString();

    // 解析+テストジョブ
    expect(
      yml.contains('analyze_test:'),
      isTrue,
      reason: '解析/テストジョブ(analyze_test)が存在すること',
    );
    expect(
      yml.contains('flutter analyze'),
      isTrue,
      reason: 'flutter analyze ステップが含まれること',
    );
    expect(
      yml.contains('flutter test'),
      isTrue,
      reason: 'flutter test ステップが含まれること',
    );

    // Android デバッグビルドジョブ
    expect(
      yml.contains('build_android:'),
      isTrue,
      reason: 'Android ビルドジョブ(build_android)が存在すること',
    );

    // iOS シミュレータビルドジョブ
    expect(
      yml.contains('build_ios:'),
      isTrue,
      reason: 'iOS ビルドジョブ(build_ios)が存在すること',
    );
  });
}

File: src/ci_analyze_build_jobs.test.ts
import { describe, it, expect } from 'vitest';
import { readFileSync } from 'node:fs';

describe('CI: analyze/test/build ジョブの存在検証', () => {
  const path = '.github/workflows/flutter-ci.yml';
  const t = readFileSync(path, 'utf-8');

  it('analyze_test ジョブが存在し、analyze/test を実行する', () => {
    expect(t).toContain('analyze_test:');
    expect(t).toMatch(/flutter analyze/);
    expect(t).toMatch(/flutter test --no-pub/);
  });

  it('build_android ジョブが存在し、必要なら flutter create を行い、APK をビルドする', () => {
    expect(t).toContain('build_android:');
    expect(t).toMatch(/flutter create --org dev\.minamidenshiimanaka \./);
    expect(t).toMatch(/flutter build apk --debug/);
    expect(t).toMatch(/actions\/upload-artifact@v4/);
  });

  it('build_ios ジョブが存在し、シミュレータ向けビルドを行う', () => {
    expect(t).toContain('build_ios:');
    expect(t).toMatch(/flutter build ios --simulator --no-codesign/);
    expect(t).toMatch(/actions\/upload-artifact@v4/);
  });
});
